### ISSUE TITLE
showConsoleCallLocation always gives undefined for the source file

### ISSUE THREAD

**Issue by joma74** [hbenl/vscode-firefox-debug#51]
vscode: 1.16.1
Debugger for Firefox: 0.16.0
Firefox: 55.02
OS: Ubuntu 14.04.5 LTS

See below for the mentioned unexcpected output
Console Firefox
```
SUCCESS BitSet can set at certain position  debug.js:15:3
Skipped 2 tests  debug.js:6:23
```
DEBUG CONSOLE Vs Code
```
SUCCESS BitSet can set at certain position (undefined:15:3)
Skipped 2 tests (undefined:6:23)
```
Just want to add a big thank to you, keep up the good work!

**Comment by hbenl** [comment]
Is the source `debug.js` mapped to a local file? In other words: if you open `debug.js` in VS Code and hover over the tab for the file, does the tooltip show a local file path or a url? And: can you set breakpoints in that file and hit them?

> Just want to add a big thank to you, keep up the good work!

Thanks :)

**Comment by joma74** [comment]
Hey,
Thank you for hinting me in the right direction :) The launch.json mapping of the source doing console logging was not mapped. 

Longer version and context for anyone hitting this issue: My setup in the context of this issue is testing via karma. My vscode launch.json configuration before writing this issue was 
```
...
{
            "type": "firefox",
            "request": "attach",
            "name": "Attach Karma Firefox",
            "showConsoleCallLocation": true,
            "port": 9222,
            "pathMappings": [
                { "url": "webpack:///", "path": "${workspaceRoot}/" }
            ]
}
...
```
karma uses webpack behind it's curtains, but ONLY for sources under test. These are served via `webpack://`. Excluded are other sources like karma.js or debug.js(both from the karma node module's  `static` folder), which are served directly via `http://`. So i amended my launch.json configuration to
```
...
"pathMappings": [
                    { "url": "webpack:///", "path": "${workspaceRoot}/" },
                    { "url": "http://localhost:9876", "path": "${workspaceRoot}/node_modules/karma/static" }
                ]
...
```
Note that `9876` is my port configuration from the karma config not shown here.
And, voila, `showConsoleCallLocation` works as expected
```
SUCCESS BitSet can set bit at position (/home/joma/entwicklung/nodews/webpack-starter-101/node_modules/karma/static/debug.js:15:3)
Skipped 0 tests (/home/joma/entwicklung/nodews/webpack-starter-101/node_modules/karma/static/debug.js:6:23)
```

@hbenl Three follow up questions if you don't mind
No1: Navigating to debug.js via VS Code in a first tab and setting a breakpoint did not get it active(being attached to the debugging session). But on clicking the link displayed via your console debug output a second tab opened with debug.js as the same source! And both tooltips over those tabs show the same path to the source! Now things get a little bit more weird - setting a breakpoint on the second source got immediately active. But execution hit the breakpoint in the source of the first tab. Hm. Any advice on how to coerce those?

No2:  Is there anything smarter one can do regarding pathMappings for non-webpack/non-bundler  sources? Cause those are very probably quite distributed over different node-modules. 

No3: Related to No2 - Curious if an addition of the same url with an alternate path or wildcard would work regarding your plugin e.g.
```
...
"pathMappings": [
                    { "url": "webpack:///", "path": "${workspaceRoot}/" },
                    { "url": "http://localhost:9876", "path": "${workspaceRoot}/node_modules/karma/static" },
                    { "url": "http://localhost:9876", "path": "${workspaceRoot}/node_modules/someOtherPath" },
                    { "url": "http://localhost:9876", "path": "${workspaceRoot}/src/main/js/**/*.js" }
                ]
...
```

**Comment by hbenl** [comment]
Concerning No. 1: this may have been the result of a bug in VS Code that was fixed in 1.17, so please try again.

No. 2: That really depends on how karma does the mapping from URLs to local paths...

No. 3: no, this won't work: the debug adapter uses the first mapping that matches the URL, so all subsequent mappings with the same URL would simply be ignored.

FYI: in VS Code 1.17, you shouldn't need `showConsoleCallLocation` anymore because VS Code now has a better way of showing source locations in the debug console.

**Comment by joma74** [comment]
No.1: 1.17 affirmative works better
No.2 and 3: ok
Thanks again, closing this issues
